gno.land/pkg/gnoweb/static/js/renderer.js

```diff
@@ -7,12 +7,87 @@ function renderUsernames(raw) {
   return raw.replace(/( |\n)@([_a-z0-9]{5,16})/, "$1[@$2](/r/demo/users:$2)");
 }
 
+// Create text nodes for the quotes with class names
+const createQuoteNode = (quote) => {
+  const node = document.createElement("a");
+  node.textContent = quote;
+  node.className = "hljs-string";
+  return node;
+};
+
 function parseContent(source, isCode) {
   if (isCode) {
+    // replace &amp; with & in code blocks
+    source = source.replace(/&amp;/g, "&");
+
     const highlightedCode = hljs.highlightAuto(source).value;
+
+    // Split the highlighted code into lines
+    const lines = highlightedCode.split('\n');
+
+    // Add line numbers to each line
+    const numberedLines = lines.map((line, index) => {
+      return `<asp class="number">${index + 1}</asp> ${line}`;
+    });
+
+    // Join the lines back into a single string
+    const numberedCode = numberedLines.join('\n');
+
+    const parser = new DOMParser();
+    const doc = parser.parseFromString(numberedCode, "text/html");
+
+    // get all span nodes of class hljs-keyword and a value of 'import'
+    const nodes = doc.querySelectorAll("span.hljs-keyword");
+    for (const node of nodes) {
+      if (node.textContent === "import") {
+        let nextNode = node.nextSibling;
+        while (true) {
```

Comment:
This is ugly. Can we tackle this in another PR, and focus on the two above changes for now?

I'm not convinced this should happen in the javascript code; _especially_ with regexes like these.

Reply:
I removed this feature in the commit: https://github.com/gnolang/gno/pull/2757/commits/a92ba96f6e8350b25e431a5bc3ec2a32860424ad